lua/lint/linters/checkpatch.lua

```diff
@@ -11,8 +15,39 @@ local severity_map = {
   ['CHECK'] = vim.diagnostic.severity.INFO,
 }
 
+local function locate_checkpatch()
+  local current_file = vim.api.nvim_buf_get_name(0)
+  local dir = current_file ~= '' and vim.fs.dirname(current_file) or vim.fn.getcwd()
+
+  -- FIXME: switch to using vim.fs.root() once minimal version is bumped to 0.10.
+  local git_dir = vim.fs.find('.git', {
+    path = dir,
+    upward = true,
+    type = 'directory',
+    limit = 1,
+  })[1]
+
+  if git_dir then
+    local project_root = vim.fs.dirname(git_dir)
+    local tool_path = vim.fs.joinpath(project_root, 'scripts', tool_name)
+    if vim.fn.executable(tool_path) == 1 then
+      return tool_path
+    end
+  end
+
+  if vim.fn.executable(tool_name) == 1 then
```

Comment:
Could you change this to only check `./scripts/checkpath.pl`? Project root dir detection is out of scope for nvim-lint as managing root heuristics that fit everyone would be too much of a hassle to maintain.  Users can provide a custom `cwd` as part of the `try_lint` call if `cwd != project_root`. 

See also:

- https://github.com/neovim/neovim/issues/34622
- https://github.com/mfussenegger/nvim-lint/issues/482#issuecomment-1999185606